app/models/user.rb

```diff
@@ -13,7 +15,11 @@ class User < ActiveRecord::Base
   validates :password, confirmation: true, presence: true,
             length: { minimum: 3 }
   validates :password_confirmation, presence: true
-  validates :email, uniqueness: true, presence: true
+  validates :email, uniqueness: true, presence: true,
+            format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/ }
+  validates :locale, presence: true,
+            inclusion: { in: I18n.available_locales.map(&:to_s),
+                         message: 'Выберите локаль из выпадающего списка.' }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.